realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -516,30 +515,11 @@ String getAccessToken(final AuthenticationServer authServer, String refreshToken
                         getUser().setRefreshToken(newRefreshToken);
                     }
                 } catch (JSONException e) {
-                    RealmLog.error(e,"Session[%s]: Can not parse the refresh_token into a valid JSONObject: ", configuration.getPath());
+                    RealmLog.error(e, "Session[%s]: Can not parse the refresh_token into a valid JSONObject: ", configuration.getPath());
                 }
             }
-            if (!onGoingAccessTokenQuery.getAndSet(true)) {
-                if (NetworkStateReceiver.isOnline(SyncObjectServerFacade.getApplicationContext())) {
-                    authenticateRealm(authServer);
-
-                } else {
-                    // Wait for connection to become available, before trying again.
-                    // The Session might potentially stay in this state for the lifetime of the application.
-                    // This is acceptable.
-                    networkListener = new NetworkStateReceiver.ConnectionListener() {
-                        @Override
-                        public void onChange(boolean connectionAvailable) {
-                            if (connectionAvailable) {
-                                if (!onGoingAccessTokenQuery.getAndSet(true)) {
-                                    authenticateRealm(authServer);
-                                }
-                                NetworkStateReceiver.removeListener(this);
-                            }
-                        }
-                    };
-                    NetworkStateReceiver.addListener(networkListener);
```

Comment:
So this logic here is being triggered indirectly by Object Store,  right?

Reply:
Yes, calling reconnect from the SyncManger should hint to the OS to call `bindSessionWithConfig` to obtain an `access_token`